wandb/integration/xgboost/xgboost.py

```diff
@@ -0,0 +1,181 @@
+        UserWarning,
+        stacklevel=2,
+    )
+
+    def callback(env):
+        for k, v in env.evaluation_result_list:
+            wandb.log({k: v}, commit=False)
+        wandb.log({})
+
+    return callback
+
+
+class WandbCallback(xgb.callback.TrainingCallback):
+    """`WandbCallback` automatically integrates XGBoost with wandb.
+    
+    Arguments:
+        log_model: (boolean) if True save and upload the model to Weights & Biases Artifacts
+        log_feature_importance: (boolean) if True log a feature importance bar plot
+        importance_type: (str) one of {weight, gain, cover, total_gain, total_cover} for tree model. weight for linear model. 
+        define_metric: (boolean) if True (default) capture model performance at the best step, instead of the last step, of training in your `wandb.summary`.
+        
+    Passing `WandbCallback` to XGBoost will:
+    
+    - log the booster model configuration to Weights & Biases
+    - log evaluation metrics collected by XGBoost, such as rmse, accuracy etc to Weights & Biases
+    - log training metric collected by XGBoost (if you provide training data to eval_set)
+    - log the best score and the best iteration
+    - save and upload your trained model to to Weights & Biases Artifacts (when `log_model = True`)
+    - log feature importance plot when `log_feature_importance=True` (default).
+    - Capture the best eval metric in `wandb.summary` when `define_metric=True` (default).
+
+    Example:
+        ```python
+        bst_params = dict(
+            objective ='reg:squarederror',
+            colsample_bytree = 0.3,
+            learning_rate = 0.1,
+            max_depth = 5,
+            alpha = 10,
+            n_estimators = 10,
+            tree_method = 'hist'
+        )
+
+        xg_reg = xgb.XGBRegressor(**bst_params)
+        xg_reg.fit(X_train, 
+            y_train, 
+            eval_set=[(X_test, y_test)],
+            callbacks=[WandbCallback()])
+        )
+        ```
+    """
+
+    def __init__(
+        self,
+        log_model: bool = False,
+        log_feature_importance: bool = True,
+        importance_type: str = "gain",
+        define_metric: bool = True,
+    ):
+
+        if wandb.run is None:
+            raise wandb.Error("You must call wandb.init() before WandbCallback()")
+
+        self.log_model = log_model
+        self.log_feature_importance = log_feature_importance
+        self.importance_type = importance_type
+        self.define_metric = define_metric
+
+    def before_training(self, model):
+        """Run before training is finished"""
+
+        # Update W&B config
+        config = model.save_config()
+        wandb.config.update(json.loads(config))
+
+        return model
+
+    def after_training(self, model):
+        """Run after training is finished."""
+
+        # Log the booster model as artifacts
+        if self.log_model:
+            self._log_model_as_artifact(model)
```

Comment:
can you use log_model in your yea test

Reply:
I can use `log_model` in my yea test but should I test it using yea logic?